web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/AttrsScope.kt

```diff
@@ -3,34 +3,30 @@ package org.jetbrains.compose.web.attributes
 import androidx.compose.runtime.DisposableEffectResult
 import androidx.compose.runtime.DisposableEffectScope
 import org.jetbrains.compose.web.css.StyleBuilder
-import org.jetbrains.compose.web.css.StyleBuilderImpl
+import org.jetbrains.compose.web.css.StyleScopeBuilder
+import org.jetbrains.compose.web.css.StyleScope
 import org.w3c.dom.Element
 import org.w3c.dom.HTMLElement
 
 /**
- * [AttrsBuilder] is a class that is used (as a builder context, that is as AttrsBuilder<T>.() -> Unit)
+ * [AttrsScope] is a class that is used (as a builder context, that is as AttrsBuilder<T>.() -> Unit)
  * in all DOM-element creating API calls. It's used for adding attributes to the element created,
  * adding inline style values (via [style]) and attaching events to the element (since AttrsBuilder
- * is an [EventsListenerBuilder])
+ * is an [EventsListenerScope])
  *
  * In that aspect the most important method is [attr]. Setting the most frequently attributes, like [id], [tabIndex]
  * are extracted to a separate methods.
  *
  */
-open class AttrsBuilder<TElement : Element> : EventsListenerBuilder() {
-    internal val attributesMap = mutableMapOf<String, String>()
-    internal val styleBuilder = StyleBuilderImpl()
-
-    internal val propertyUpdates = mutableListOf<Pair<(Element, Any) -> Unit, Any>>()
-    internal var refEffect: (DisposableEffectScope.(TElement) -> DisposableEffectResult)? = null
-
-    internal var inputControlledValueSet = false
-    internal var inputDefaultValueSet = false
-    internal var inputControlledCheckedSet = false
-    internal var inputDefaultCheckedSet = false
-
+interface AttrsScope<TElement : Element>: EventsListenerScope {
+    val attributesMap: Map<String, String>
```

Comment:
Do these public val need to be in the interface? 
Can they be only in the implementation? 

they were internal, so it seems all of them are exposed to external code via functions in a controllable manner. Am I missing something?

Reply:
I'll double check, but I only moved things into the interface one by one when I've discovered some outside usage of them (i.e. extension functions withing compose-web).

Reply:
I've checked, and because they were internal, but on the concrete implementation class used by all the builder lambda arguments elsewhere they were fine. Now that all builders are using the interface, some internal code still needs them accessible and as such they cannot just be internal vals on the implementation as builders do not see it.

Reply:
However such usage only occurs in your internal test suites so far.